x-pack/plugins/security/public/nav_control/nav_control_component.tsx

```diff
@@ -93,17 +92,18 @@ export const SecurityNavControl: FunctionComponent<SecurityNavControlProps> = ({
     items.push(...userMenuLinkMenuItems);
   }
 
-  if (!isAnonymous) {
-    const hasCustomProfileLinks = userMenuLinks.some(({ setAsProfile }) => setAsProfile === true);
+  const isAnonymous = currentUser.value ? isUserAnonymous(currentUser.value) : false;
+  const hasCustomProfileLinks = userMenuLinks.some(({ setAsProfile }) => setAsProfile === true);
+
+  if (!isAnonymous && !hasCustomProfileLinks) {
```

Comment:
__issue:__ while testing this change I discovered [that we rely on the user's roles](https://github.com/elastic/kibana/blob/b72601c283bd9982ef2ed415b40e3cef179e19ac/x-pack/plugins/cloud/public/plugin.tsx#L195-L219) (specifically `superuser` role) to determine whether we should show Cloud specific links or not (added in https://github.com/elastic/kibana/pull/97870). If we keep this Cloud plugin logic as is, after this PR non-Cloud managed superusers won't be able to edit their profiles in Kibana (we'll be rendering a single `Edit profile` link leading to Cloud UI instead of `Profile` and `Preferences` links we had previously).

@legrego do I understand correctly that we can now switch [this code](https://github.com/elastic/kibana/blob/b72601c283bd9982ef2ed415b40e3cef179e19ac/x-pack/plugins/cloud/public/plugin.tsx#L195-L219) to rely on [`elastic_cloud_user` user property](https://github.com/elastic/kibana/blob/9ea87308866d298bf55ce7b8a667cb89c6587e4a/x-pack/plugins/security/server/authentication/providers/base.ts#L142-L145) instead or is there still a legitimate reason for non-Cloud managed superusers to see Cloud specific links?

Reply:
> @legrego do I understand correctly that we can now switch [this code](https://github.com/elastic/kibana/blob/b72601c283bd9982ef2ed415b40e3cef179e19ac/x-pack/plugins/cloud/public/plugin.tsx#L195-L219) to rely on [`elastic_cloud_user` user property](https://github.com/elastic/kibana/blob/9ea87308866d298bf55ce7b8a667cb89c6587e4a/x-pack/plugins/security/server/authentication/providers/base.ts#L142-L145) instead or is there still a legitimate reason for non-Cloud managed superusers to see Cloud specific links?

That's correct, we should switch this code to rely on `elastic_cloud_user` instead of the role check. The `superuser` check was a hard-coded workaround because we didn't have a way to reliably determine which users were cloud users back then.

Reply:
Great, thanks for confirming, Larry.